src/coreclr/nativeaot/System.Private.CoreLib/src/Internal/Reflection/Extensions/NonPortable/CustomAttributeInstantiator.cs

```diff
@@ -144,8 +144,7 @@ public static Attribute Instantiate(this CustomAttributeData cad)
                 IList<CustomAttributeTypedArgument>? typedElements = (IList<CustomAttributeTypedArgument>?)(typedArgument.Value);
                 if (typedElements == null)
                     return null;
-                Type? elementType = argumentType.GetElementType();
-                Array array = Array.CreateInstance(elementType, typedElements.Count);
+                Array array = Array.CreateInstanceFromArrayType(argumentType, typedElements.Count);
```

Comment:
I think the `UnconditionalSuppressMessage` attribute at the top of this method can now be deleted - it was suppressing a warning caused by the Array.CreateInstance. I don't see anything else in this method that would generate this warning.